packages/gradle/src/plugin/dependencies.ts

```diff
@@ -2,94 +2,66 @@ import {
-          (project) => project.root === targetProjectRoot
-        )?.name;
-        if (targetProjectName) {
-          const dependency: RawProjectGraphDependency = {
-            source: projectName as string,
-            target: targetProjectName as string,
-            type: DependencyType.static,
-            sourceFile: gradleFile,
-          };
-          validateDependency(dependency, context);
-          dependencies.add(dependency);
-        }
-      });
-    }
-    gradleProjectToChildProjects.get(gradleProject)?.forEach((childProject) => {
-      if (childProject) {
-        const dependency: RawProjectGraphDependency = {
-          source: projectName as string,
-          target: childProject,
-          type: DependencyType.static,
-          sourceFile: gradleFile,
-        };
-        validateDependency(dependency, context);
-        dependencies.add(dependency);
-      }
-    });
-  }
-
-  const gradleDependenciesEnd = performance.mark('gradleDependencies:end');
-  performance.measure(
-    'gradleDependencies',
-    gradleDependenciesStart.name,
-    gradleDependenciesEnd.name
+  const files = await globWithWorkspaceContext(
+    workspaceRoot,
+    Array.from(GRALDEW_FILES)
   );
+  const { gradlewFiles } = splitConfigFiles(files);
+  await populateProjectGraph(context.workspaceRoot, gradlewFiles, options);
+  const { dependencies: dependenciesFromReport } =
+    getCurrentProjectGraphReport();
 
-  return Array.from(dependencies);
-};
-
-function findGradleFiles(fileMap: FileMap): string[] {
-  const gradleFiles: string[] = [];
-
-  for (const [_, files] of Object.entries(fileMap.projectFileMap)) {
-    for (const file of files) {
-      if (GRADLE_BUILD_FILES.has(basename(file.file))) {
-        gradleFiles.push(file.file);
+  const dependencies: Array<StaticDependency> = [];
+  dependenciesFromReport.forEach((dependencyFromPlugin: StaticDependency) => {
+    try {
+      const source =
+        relative(workspaceRoot, dependencyFromPlugin.source) || '.';
+      const sourceProjectName =
+        Object.values(context.projects).find(
+          (project) => source === project.root
+        )?.name ?? dependencyFromPlugin.source;
+      const target =
+        relative(workspaceRoot, dependencyFromPlugin.target) || '.';
+      const targetProjectName =
+        Object.values(context.projects).find(
+          (project) => target === project.root
+        )?.name ?? dependencyFromPlugin.target;
+      if (!sourceProjectName || !targetProjectName) {
+        return;
       }
-    }
-  }
+      const dependency: StaticDependency = {
+        source: sourceProjectName,
+        target: targetProjectName,
+        type: DependencyType.static,
+        sourceFile: normalizePath(
+          relative(workspaceRoot, dependencyFromPlugin.sourceFile)
+        ),
+      };
+      validateDependency(dependency, context);
+      dependencies.push(dependency);
+    } catch {} // ignore invalid dependencies
```

Comment:
We should at least warn if we really need it.

Reply:
changed